examples/gno.land/p/demo/microblog/microblog.gno

```diff
@@ -0,0 +1,198 @@
+}
+
+func NewMicroblog(title string, prefix string) (m *Microblog) {
+	return &Microblog{
+		Title:  title,
+		Prefix: prefix,
+		Pages:  avl.Tree{},
+	}
+}
+
+func (m *Microblog) GetPages() []*Page {
+	var (
+		pages = make([]*Page, m.Pages.Size())
+		index = 0
+	)
+
+	m.Pages.Iterate("", "", func(key string, value interface{}) bool {
+		pages[index] = value.(*Page)
+		index++
+		return false
+	})
+
+	sort.Sort(ByLastPosted(pages))
+
+	return pages
+}
+
+func (m *Microblog) RenderHome() string {
+	output := ufmt.Sprintf("# %s\n\n", m.Title)
+	output += "# pages\n\n"
+
+	for _, page := range m.GetPages() {
+		output += ufmt.Sprintf("- [%s](%s%s)\n", page.Author.String(), m.Prefix, page.Author.String())
+	}
+
+	return output
+}
+
+func (m *Microblog) RenderUser(user string) string {
+	silo, found := m.Pages.Get(user)
+	if !found {
+		return StatusNotFound
+	}
+
+	return (silo.(*Page)).String()
+}
+
+func (m *Microblog) Render(path string) string {
+	parts := strings.Split(path, "/")
+
+	isHome := path == ""
+	isUser := len(parts) == 1
+
+	switch {
+	case isHome:
+		return m.RenderHome()
+
+	case isUser:
+		return m.RenderUser(parts[0])
+	}
+
+	return StatusNotFound
+}
+
+func (m *Microblog) newPage() error {
+	author := std.GetOrigCaller()
+	_, found := m.Pages.Get(author.String())
+	if found {
+		return errors.New("author already exists")
+	}
+
+	m.Pages.Set(author.String(), &Page{
+		Author:    author,
+		CreatedAt: time.Now(),
+	})
+	return nil
+}
+
+func (m *Microblog) NewPost(text string) error {
+	author := std.GetOrigCaller()
+	_, found := m.Pages.Get(author.String())
+	if !found {
+		// make a new page for the new author
+		if err := m.newPage(); err != nil {
+			return err
+		}
+	}
+
+	page, err := m.GetPage(author.String())
+	if err != nil {
+		return err
+	}
+	return page.NewPost(text)
+}
+
+func (m *Microblog) GetPage(author string) (*Page, error) {
+	silo, found := m.Pages.Get(author)
+	if !found {
+		return nil, ErrNotFound
+	}
+	return silo.(*Page), nil
+}
+
+type Page struct {
+	ID         int
+	Author     std.Address
+	CreatedAt  time.Time
+	LastPosted time.Time
+	Posts      avl.Tree // time -> Post
+}
+
+// ByLastPosted implements sort.Interface for []Page based on
+// the LastPosted field.
+type ByLastPosted []*Page
+
+func (a ByLastPosted) Len() int           { return len(a) }
+func (a ByLastPosted) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ByLastPosted) Less(i, j int) bool { return a[i].LastPosted.After(a[j].LastPosted) }
+
+func (p *Page) String() string {
+	o := ufmt.Sprintf("# %s\n\n", p.Author)
+	o += ufmt.Sprintf("joined %s, last updated %s\n\n", p.CreatedAt.Format("2006-02-01"), p.LastPosted.Format("2006-02-01"))
+	o += "## posts\n\n"
+	for _, u := range p.GetPosts() {
+		o += u.String() + "\n\n"
+	}
+	return o
+}
+
+func (p *Page) NewPost(text string) error {
+	now := time.Now()
+	p.LastPosted = now
+	p.Posts.Set(ufmt.Sprintf("%s%d", now, p.Posts.Size()), &Post{
```

Comment:
```suggestion
	p.Posts.Set(ufmt.Sprintf("%s%d", now.Format(time.RFC3339), p.Posts.Size()), &Post{
```

While it's unlikely that the format of `time.Time.String()` will ever change, it's not meant to be used for serialization, so I would say instead use a format on a known format, which still sorts alphabetically?

Reply:
Yes, good idea. Fixed with 7da151fa